crates/indexer/src/helper.rs

```diff
@@ -0,0 +1,191 @@
+use anyhow::{anyhow, Result};
+use gw_common::H256;
+use gw_types::packed::LogItem;
+use gw_types::prelude::*;
+use std::{convert::TryInto, usize};
+
+pub const GW_LOG_SUDT_TRANSFER: u8 = 0x0;
+pub const GW_LOG_SUDT_PAY_FEE: u8 = 0x1;
+pub const GW_LOG_POLYJUICE_SYSTEM: u8 = 0x2;
+pub const GW_LOG_POLYJUICE_USER: u8 = 0x3;
+#[derive(Default, Debug)]
+pub struct PolyjuiceArgs {
+    pub is_create: bool,
+    pub gas_limit: u64,
+    pub gas_price: u128,
+    pub value: u128,
+    pub input: Option<Vec<u8>>,
+}
+
+impl PolyjuiceArgs {
+    // https://github.com/nervosnetwork/godwoken-polyjuice/blob/v0.6.0-rc1/polyjuice-tests/src/helper.rs#L322
+    pub fn decode(args: &[u8]) -> anyhow::Result<Self> {
+        let is_create = args[7] == 3u8;
+        let gas_limit = u64::from_le_bytes(args[8..16].try_into()?);
+        let gas_price = u128::from_le_bytes(args[16..32].try_into()?);
+        let value = u128::from_le_bytes(args[32..48].try_into()?);
+        let input_size = u32::from_le_bytes(args[48..52].try_into()?);
+        let input: Vec<u8> = args[52..(52 + input_size as usize)].to_vec();
```

Comment:
Potential out of bound attacks, please check the args len and input_size first

Reply:
`GW_MAX_L2TX_ARGS_SIZE`, known as the max size of `PolyjuiceArgs` is 128KB.

It is defined in [`godwoken-scripts/c/gw_def.h`](https://github.com/nervosnetwork/godwoken-scripts/blob/fb69b35aaadef70dba162331bec3e2f25517714f/c/gw_def.h).

Reply:
updated